edx_exams/apps/api/v1/views.py

```diff
@@ -271,3 +274,116 @@ class ProctoringProvidersView(ListAPIView):
     model = ProctoringProvider
     serializer_class = ProctoringProviderSerializer
     queryset = ProctoringProvider.objects.all()
+
+
+class ExamAccessTicketsView(APIView):
+    """
+    View to create signed exam access tickets for a specific user and exam.
+
+    Given an exam_id and user (in request), this view will either grant access
+    as an exam access ticket or not grant access. Access is granted if there is an
+    existing exam attempt or if the exam is past its due date.
+
+    HTTP GET
+    Provides an Exam Access Ticket as a cookie if access is granted.
+    **GET data Parameters**
+        * exam_id: This is the id of the exam that the user is requesting access to
+    **Exceptions**
+        * HTTP_403_FORBIDDEN
+    """
+
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (StaffUserOrReadOnlyPermissions,)
+
+    @classmethod
+    def get_current_exam_attempt(cls, user, exam):
+        """
+        Given a user and exam, get the user's latest exam attempt, if exists.
+        """
+        try:
+            exam_attempt = ExamAttempt.objects.filter(user=user, exam=exam).latest('created')
+        except ObjectDoesNotExist:
+            exam_attempt = None
+        return exam_attempt
+
+    @classmethod
+    def set_expiration_window(cls, exam_attempt, default_exp_seconds):
+        """
+        Set exam access ticket expiration window.
+
+        Use default exp window or the time to the end of the exam attempt,
+        whichever is shorter.
+        """
+        secs_remaining = get_exam_attempt_time_remaining(exam_attempt.start_time,
+                                                         exam_attempt.allowed_time_limit_mins)
+
+        # If not past due and attempt time remaining is less than default, set window to remaining
+        if 0 <= secs_remaining < default_exp_seconds:
+            exp_window = secs_remaining
+        else:
+            exp_window = default_exp_seconds
+
+        return exp_window
+
+    @classmethod
+    def get_response(cls, exam, user):
+        """
+        Get response, with exam access ticket if access granted.
+
+        403 error if access is not granted.
+        """
+        claims = {"course_id": exam.course_id, "content_id": exam.content_id}
+        expiration_window = 60
+        exam_attempt = cls.get_current_exam_attempt(user, exam)
+        cookie_name = "exam_access_ticket"
+
+        data = {}
+        grant_access = True
+        response_status = status.HTTP_200_OK
+
+        # If exam attempt exists for user, then grant exam access
+        if exam_attempt is not None:
+            log.info("Creating exam access ticket for attempt")
+            # If started, determine how to set expiration window
+            # todo: should exam attempt status be started or something else?
+            if exam_attempt.status == "started":
```

Comment:
Question: is this the correct check for the exam attempt status?

Reply:
yes, but we should probably introduce a constant/enum for attempt statuses. Example https://github.com/openedx/edx-proctoring/blob/0ae746c253db5910cd6f91abf60e19771102130c/edx_proctoring/statuses.py#L8

Reply:
That makes sense, will do.